tests/ut/language/parser/test_scope_parsing.py

```
@@ -594,5 +594,261 @@ def main(
+        """Entries other than ``pl.auto_chunk`` / ``pl.split(...)`` are rejected."""
+        with pytest.raises(ParserSyntaxError, match="Unsupported entry"):
+
+            @pl.function
+            def bad(a: pl.Tensor[[64], pl.FP32]) -> pl.Tensor[[64], pl.FP32]:
+                for i in pl.spmd(4, optimizations=[pl.range]):  # type: ignore[list-item]
+                    _ = i
+                return a
+
+    def test_spmd_rejects_duplicate_split(self):
+        """Duplicate ``pl.split(...)`` in the list is rejected."""
+        with pytest.raises(ParserSyntaxError, match="Duplicate 'pl.split"):
```

Ruff (0.15.13)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)